src/component.js

```diff
@@ -0,0 +1,51 @@
+import React from "react";
+import NProgress from "nprogress";
+import Router from "next/router";
+import NProgressStyles from "./styles";
+
+class NProgressContainer extends React.Component {
+  constructor (props) {
+    super(props)
+    this.timer = null;
+  }
+
+  componentDidMount () {
+    if (this.props.nprogressOptions) {
+      NProgress.configure(this.props.nprogressOptions);
+    }
+
+    Router.onRouteChangeStart = () => {
+      this.clearTimer();
+      this.timer = setTimeout(NProgress.start, this.props.showAfterMs || 300);
+    };
+
+    Router.onRouteChangeComplete = () => {
+      NProgress.done();
+      this.clearTimer();
+    };
+
+    Router.onRouteChangeError = () => {
+      NProgress.done();
+      this.clearTimer();
+    };
+  }
+
+  clearTimer () {
+    if (this.timer) {
+      clearTimeout(this.timer);
```

Comment:
This could be done without checking for `this.timer`.

Do the clearTimeout and we can avoid an extra method, the clearTimeout function will do nothing if the timer ID is invalid.